### PR TITLE
feat: distribution kyc 2

### PR DESCRIPTION
**Pull request by ice-hades**

### CODE REVIEW

File: src/api/auth/types.ts [hunk 1]
```diff
@@ -27,8 +27,16 @@ type EthDistributionKyc = {
   };
 };
 
+type EthPeriodicDistributionKyc = {
+  'social3-kyc': {
+    enabled: boolean;
```
Comment (ice-ares) [inline src/api/auth/types.ts]: what's this for?

Comment (ice-hades) [reply src/api/auth/types.ts]: removed

**Review by ice-hector — DISMISSED** [review]

**Review by ice-ares — DISMISSED** [review]

File: src/api/tokenomics/constants.ts [hunk 2]
```diff
@@ -11,4 +12,5 @@ export const SELFIE_KYC_STEP: SelfieKycStep = 1;
 export const EMOTIONS_KYC_STEP: EmotionsKycStep = 2;
 
 export const VERIFY_SOCIAL_ACCOUNT_KYC_STEP: VerifySocialAccountKycStep = 3;
-export const ETH_DISTRIBUTION_KYC_STEP: EthDistributionKycStep = 5;
+export const DISTRIBUTION_KYC_STEP: DistributionKycStep = 5;
+export const DYNAMIC_DISTRIBUTION_KYC_STEP: DynamicDistributionKycStep = 6;
```
Comment (ice-ares) [inline src/api/tokenomics/constants.ts]: remove this

File: src/api/tokenomics/types.ts [hunk 3]
```diff
@@ -78,10 +78,9 @@ export type EmotionsKycStep = 2;
 export type FaceAuthKycNumber = SelfieKycStep | EmotionsKycStep;
 
 export type VerifySocialAccountKycStep = 3;
-export type EthDistributionKycStep = 5;
-export type SocialKycStepNumber =
-  | VerifySocialAccountKycStep
-  | EthDistributionKycStep;
+export type DistributionKycStep = 5;
+export type DynamicDistributionKycStep = 6;
```
Comment (ice-ares) [inline src/api/tokenomics/types.ts]: remove this

File: src/api/tokenomics/constants.ts (hunk 2 above)
Comment (ice-hades) [reply src/api/tokenomics/constants.ts]: no. it is needed. see how it's used

File: src/api/tokenomics/types.ts (hunk 3 above)
Comment (ice-hades) [reply src/api/tokenomics/types.ts]: no. it is needed. see how it's used

File: src/store/modules/Tokenomics/sagas/startMiningSession.ts [hunk 4]
```diff
@@ -129,6 +131,20 @@ export function* startMiningSessionSaga(
           });
           return;
         }
+        const kycStep = errorData.kycSteps?.[0] ?? 0;
+        if (kycStep >= DYNAMIC_DISTRIBUTION_KYC_STEP) {
```
Comment (ice-ares) [inline src/store/modules/Tokenomics/sagas/startMiningSession.ts]: no, please iterate over the array like I asked

Comment (ice-hades) [reply src/store/modules/Tokenomics/sagas/startMiningSession.ts]: it is right here. iteration was before

File: src/api/tokenomics/constants.ts (hunk 2 above)
Comment (ice-hades) [reply src/api/tokenomics/constants.ts]: fixed

File: src/api/tokenomics/types.ts (hunk 3 above)
Comment (ice-hades) [reply src/api/tokenomics/types.ts]: fixed

File: src/store/modules/Tokenomics/sagas/startMiningSession.ts (hunk 4 above)
Comment (ice-hades) [reply src/store/modules/Tokenomics/sagas/startMiningSession.ts]: fixed

File: src/screens/SocialKycFlow/utils.ts [hunk 5]
```diff
@@ -10,5 +11,12 @@ export function kycStepToTranslationsPathPrefix(kycStep: SocialKycStepNumber) {
...
       return 'distribution_kyc';
     }
+    default: {
+      return 'distribution_kyc_dynamic';
+    }
   }
 }
+
+export function isDistributionKyc(kycStep: SocialKycStepNumber): boolean {
+  return kycStep !== VERIFY_SOCIAL_ACCOUNT_KYC_STEP;
```
Comment (ice-ares) [inline src/screens/SocialKycFlow/utils.ts:21]: Dont u also need to check if its in that array?

Comment (ice-hades) [reply src/screens/SocialKycFlow/utils.ts:21]: no. that is inside UI for social and distribution KYCs
so, if kycStep !== VERIFY_SOCIAL_ACCOUNT_KYC_STEP  means it is a distribution

**Review by ice-ares — DISMISSED** [review]

**Review by ice-orion — DISMISSED** [review]

**Review by ice-hector — DISMISSED** [review]

**Review by ice-ares — DISMISSED** [review]

File: src/store/modules/Tokenomics/sagas/startMiningSession.ts [hunk 6]
```diff
@@ -131,6 +132,19 @@ export function* startMiningSessionSaga(
         } else if (errorData.kycSteps.includes(QUIZ_KYC_STEP)) {
           yield put(QuizActions.START_OR_CONTINUE_QUIZ_FLOW.STATE.create());
           return;
+        } else {
+          const kycStep = errorData.kycSteps?.[0] ?? 0;
+          const dynamicDistributionData: ReturnType<
+            typeof dynamicDistributionDataSelector
+          > = yield select(dynamicDistributionDataSelector);
+          if (dynamicDistributionData?.some(data => data?.step === kycStep)) {
+            yield removeScreenByName('Tooltip').catch();
```
Comment (ice-orion) [inline src/store/modules/Tokenomics/sagas/startMiningSession.ts]: it is already called on line 113

Comment (ice-hades) [reply src/store/modules/Tokenomics/sagas/startMiningSession.ts]: fixed

**Review by ice-hector — APPROVED** [review]

**Review by ice-orion — APPROVED** [review]

**Pull request merged**